autoload/clever_f.vim

```diff
@@ -156,11 +156,18 @@ function! clever_f#_mark_direct(forward, count) abort
 
     let char_count = {}
     let matches = []
-    let indices = a:forward ? range(c, len(line) - 1, 1) : range(c - 2, 0, -1)
-    for i in indices
-        let ch = line[i]
-        " only matches to ASCII
-        if ch !~# '^[\x00-\x7F]$' | continue | endif
+    if a:forward 
+        let line = split(line[c - 1 : ], '\zs') " adding a limit like c+4*termcol*termlines just after `:` is a free gift in a very long line, assuming all char are less than 4 bytes.
```

Comment:
Note that this does not work since some characters are longer than 4 bytes like 👪 or 🇺🇸

Reply:
Agreed, lines can wrap. Hence the screen width is not enough, that is why I multiplied the number of columns of the screen by the number of lines in the screen.